src/core/thread/mesh_forwarder.cpp

```diff
@@ -1292,38 +1292,37 @@ otError MeshForwarder::ParseIp6UdpTcpHeader(const Message &aMessage,
                                             uint16_t &     aDestPort)
 {
     otError error = OT_ERROR_PARSE;
-    union
```

Comment:
Does this change increase stack overhead?

Reply:
I think it may. It depends on compiler/toolchain.
Will change it back.